workflow/events/event_recorder_manager.go

```diff
@@ -24,14 +27,43 @@ type eventRecorderManager struct {
 	eventRecorders map[string]record.EventRecorder
 }
 
+// customEventAggregatorFuncWithAnnotations enhances the default `EventAggregatorByReasonFunc` by
+// including annotation values as part of the event aggregation key.
+func customEventAggregatorFuncWithAnnotations(event *apiv1.Event) (string, string) {
+	var joinedAnnotationsStr string
+	includeAnnotations := env.LookupEnvStringOr("EVENT_AGGREGATION_WITH_ANNOTATIONS", "false")
+	if annotations := event.GetAnnotations(); includeAnnotations == "true" && annotations != nil {
+		var joinedAnnotations = make([]string, 0, len(annotations))
+		for _, v := range annotations {
+			joinedAnnotations = append(joinedAnnotations, v)
+		}
+		joinedAnnotationsStr = strings.Join(joinedAnnotations, "")
```

Comment:
I think there is a bug here. Annotations is a map and when iterating maps in golang the order is random. Therefore your string will be different every time.

Reply:
Good catch! Fixed.